src/soundio/sounddeviceportaudio.cpp

```diff
@@ -886,13 +888,26 @@ int SoundDevicePortAudio::callbackProcessClkRef(
             m_deviceId.debugName());
 
     //qDebug() << "SoundDevicePortAudio::callbackProcess:" << m_deviceId;
-    // Turn on TimeCritical priority for the callback thread. If we are running
-    // in Linux userland, for example, this will have no effect.
+
     if (!m_bSetThreadPriority) {
+#ifdef __LINUX__
+        // Verify if we are a thread with "real-time" policy.
+        // The audio thread on Linux should be set to SCHED_FIFO with a priority
+        // that's somewhere between 60 and 90 depending on the allowed priority
+        // ranges (some USB devices by default get assigned a priority in the
+        // 50s with some system configs).
+        if ((sched_getscheduler(0) & SCHED_FIFO) == 0) {
+            qWarning() << "Engine thread not scheduled with the real-time policy SCHED_FIFO";
```

Comment:
scanning the manpages, there is also the `SCHED_RR` RT policy. Why is that not valid? Also, we could assert that the priority is in the mentioned range by retrieving it via `sched_getparam`. Would that make sense?

Reply:
`SCHED_RR` doesn't make much sense. Realtime work needs to be finished as soon as it's available, and with `SCHED_RR` you don't have that guarantee. I don't think I've ever seen anything use it. 

And I don't think checking the realtime priority is needed. As long as the thread uses realtime scheduling it's probably good enough for most use cases. And if the user uses PipeWire as a JACK implementation without realtime having realtime priviliges, then this would use RTKit which IIRC by default limits the maximum priority to somewhere in the 20s.

Reply:
Makes sense. Thank you for jumping in here and sharing your expertise.